README.md

```diff
@@ -2,31 +2,365 @@
 
 ## Learning Goals
 
-- Learning Goal 1
-- Learning Goal 2
+- Explain constructors
+- Create constructors in Java
 
 ## Introduction
 
-A 1-2 sentence summary of what will be covered.
+A **constructor** is a specific kind of method that is called when an object is
+instantiated.  Recall that we create an instance of `Dog` with the following code snippet:
 
-## Topic 1
+```java
+new Dog()
+```
 
-Discuss the topic of the reading. Break into sections and sub-sections as
-appropriate to help with the reader's orientation with the material and future
-referencing.
+The code consists of 2 parts:
 
-## Topic 2
+- Instantiation − The `new` operator is used to create an instance of `Dog` by allocating memory
+  to store the instance variables.
 
-Discuss the topic of the reading. Break into sections and sub-sections as
-appropriate to help with the reader's orientation with the material and future
-referencing.
+- Initialization − The `new` operator is followed by a call to a constructor `Dog()`.  The
+  constructor is a method that initializes the new object by assigning a value to each instance variable.
 
-## Conclusion
+General rules about constructor methods:
+
+- A constructor has the same name as the class.
+- A constructor can have parameters.
+- A constructor does not have a return type, not even void.
+- A constructor implicitly returns a reference to the newly created object.  Do not use an explicit return statement.
+- A class may have multiple constructors, called **constructor overloading**, as long as the number or type of parameters differ.
+
+## Default Constructor 
+
+![noargs signature](https://curriculum-content.s3.amazonaws.com/6676/java-methods/noargs_signature.png)
+
+A constructor with no arguments such as `Dog()` is called the *default* constructor or *no-args* constructor.
+
+- If you *do not* define a constructor, Java implicitly defines a default no-args constructor.
+- If you *do* define a constructor with or without parameters, Java *does not* generate a default constructor.
+
+The default constructor for `Dog` initializes instance variables to default values
+based on their data type:
+
+```java
+
+// Constructor
+public Dog() {
```

Comment:
Suggestion: Show the private instance variables above too or show the `Dog` class again in the introduction

Reply:
ok